src/system/group.cr

```diff
@@ -0,0 +1,50 @@
+require "crystal/system/group"
+
+class System::Group
+  # Raised on group lookup failure.
+  class NotFoundError < Exception
+  end
+
+  extend Crystal::System::Group
+
+  getter name : String
+  getter id : String
+  getter members : Array(String)
+
+  def_equals_and_hash @id
+
+  private def initialize(@name, @id, @members)
+  end
+
+  # Returns the group associated with the given name.
+  #
+  # Raises `NotFoundError` if no such group exists.
+  def self.find_by(*, name)
```

Comment:
Can we add a type restriction to `name`, like `name : String`? Same with all other methods that lack a type restriction. And put a return type too.